### ISSUE TITLE
Read data in Chapter 4

### ISSUE THREAD

**Issue by benjamin-coppel** [hadley/mastering-shiny#595]
When you want to download *NEISS data*, the url of the repo doesn´t work.

I think that the url was update. 

You need to put this url in order to fix the **download**  function

`download <- function(name){
  url <- "https://github.com/hadley/mastering-shiny/tree/master/neiss/"
  download.file(paste0(url, name), paste0("neiss/", name), quiet = TRUE)
}`

**Comment by benjamin-coppel** [comment]
For some reason, when I download the data using the function and then read it, the data looks so weird.

I have to download manually

**Comment by samlin829** [comment]
Hi everyone, as of 18/Dec/2022 the url to the neiss dataset is `https://github.com/hadley/mastering-shiny/blob/main/neiss/`. Confirmed on my machine to download the data programmatically.

**Comment by nmolanog** [comment]
> 

I confirm I have the same issue

**Comment by nmolanog** [comment]
> For some reason, when I download the data using the function and then read it, the data looks so weird.
> 
> I have to download manually

I have the same issue

**Comment by benjamin-coppel** [comment]
@carehu My "non-fancy" solution was download manually the files, put them in a folder and then read the data from this new folder in order to follow the example in the book.

**Comment by carehu** [comment]
> @carehu My "non-fancy" solution was download manually the files, put them in a folder and then read the data from this new folder in order to follow the example in the book.

Thank you! For products & population data I did like you explained and is working. But for the injuries data it still doesn't work. I have used the R.utils package to unzip the file and if I open the unzipped file in notepad it doesn't look like what is in the book. For now I think I will just skip the examples in this chapter. Thanks.

library(R.utils)
gunzip("neiss/injuries.tsv.gz", remove=FALSE)
